internal/api/resolver_mutation_studio.go

```diff
@@ -28,190 +24,284 @@ func (r *mutationResolver) getStudio(ctx context.Context, id int) (ret *models.S
+		} else {
+			parentTranslator = changesetTranslator{
+				inputMap: getNamedUpdateInputMap(ctx, updateInputField+".parent"),
+			}
 
-	// Process the base 64 encoded image string
-	if input.Image != nil {
-		imageData, err = utils.ProcessImageInput(ctx, *input.Image)
+			// The parent studio matched an existing one and the user has chosen in the UI to link and/or update it
+			dbInput.ParentUpdate, err = studioPartialFromStudioCreateInput(ctx, *input.Parent, input.ParentID, parentTranslator)
+			runParentUpdateHook = true
+		}
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	dbInput.StudioCreate, err = studioFromStudioCreateInput(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	// Start the transaction and save the studio
+	if err := r.withTxn(ctx, func(ctx context.Context) error {
+		qb := r.repository.Studio
+		studioID, err = qb.Create(ctx, dbInput)
+		if err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	newStudio, err := r.getStudio(ctx, *studioID)
+	if err != nil {
+		return nil, fmt.Errorf("finding after create: %w", err)
+	}
+
+	if runParentCreateHook {
+		r.hookExecutor.ExecutePostHooks(ctx, *newStudio.ParentID, plugin.StudioCreatePost, input, nil)
+	} else if runParentUpdateHook {
+		r.hookExecutor.ExecutePostHooks(ctx, *newStudio.ParentID, plugin.StudioUpdatePost, input, parentTranslator.getFields())
+	}
+	r.hookExecutor.ExecutePostHooks(ctx, *studioID, plugin.StudioCreatePost, input, nil)
+
+	return newStudio, nil
+}
+
+func studioFromStudioCreateInput(ctx context.Context, input StudioCreateInput) (*models.Studio, error) {
 	// Populate a new studio from the input
-	currentTime := time.Now()
 	newStudio := models.Studio{
-		Checksum:  checksum,
-		Name:      sql.NullString{String: input.Name, Valid: true},
-		CreatedAt: models.SQLiteTimestamp{Timestamp: currentTime},
-		UpdatedAt: models.SQLiteTimestamp{Timestamp: currentTime},
+		Name: input.Name,
 	}
 	if input.URL != nil {
-		newStudio.URL = sql.NullString{String: *input.URL, Valid: true}
+		newStudio.URL = *input.URL
 	}
 	if input.ParentID != nil {
-		parentID, _ := strconv.ParseInt(*input.ParentID, 10, 64)
-		newStudio.ParentID = sql.NullInt64{Int64: parentID, Valid: true}
+		parentID, _ := strconv.Atoi(*input.ParentID)
+		newStudio.ParentID = &parentID
+	}
+	if input.Details != nil {
+		newStudio.Details = *input.Details
 	}
-
 	if input.Rating100 != nil {
-		newStudio.Rating = sql.NullInt64{
-			Int64: int64(*input.Rating100),
-			Valid: true,
-		}
+		newStudio.Rating = input.Rating100
 	} else if input.Rating != nil {
-		newStudio.Rating = sql.NullInt64{
-			Int64: int64(models.Rating5To100(*input.Rating)),
-			Valid: true,
-		}
-	}
-
-	if input.Details != nil {
-		newStudio.Details = sql.NullString{String: *input.Details, Valid: true}
+		rating := models.Rating5To100(*input.Rating)
+		newStudio.Rating = &rating
 	}
 	if input.IgnoreAutoTag != nil {
 		newStudio.IgnoreAutoTag = *input.IgnoreAutoTag
 	}
 
```

Comment:
Image is now downloaded, but the saving to the db has been moved to pkg/sqlite/studio.go to avoid that duplicated block of code.
Same with saving the alias and stashid code blocks.